plugins/outputs/mqtt/sample.conf

```diff
@@ -13,8 +13,16 @@
 
   ## MQTT Topic for Producer Messages
   ## MQTT outputs send metrics to this topic format:
-  ## <topic_prefix>/<hostname>/<pluginname>/ (e.g. prefix/web01.example.com/mem)
-  topic_prefix = "telegraf"
+  ## {{ .TopicPrefix }}/{{ .Hostname }}/{{ .PluginName }}/{{ .Tag "tag_key" }}
+  ## (e.g. prefix/web01.example.com/mem/some_tag_value)
+  ## Each path segment accepts either a template placeholder, an environment variable, or a tag key
+  ## of the form `{{.Tag "tag_key_name"]]`. Empty path elements as well as special MQTT characters
+  ## (such as `+` or `#`) are invalid to form the topic name and will lead to an error.
+  ## In case a tag is missing in the metric, that path segment omitted for the final topic.
+  topic = "telegraf/{{ .Hostname }}/{{ .PluginName }}"
+
+  ## DEPRECATED: Use the topic template instead.
+  ## topic_prefix = "telegraf"
```

Comment:
@srebhan I haven't removed this, just commented and as you can see, the value `telegraf` is now part of the topic above, as a hardcoded initial value, instead of a reference to the now deprecated `topic_prefix`. This is to keep backward compatibility to previous behaviour.

I hope this is what you had in mind, or do you want me just to entirely remove lines 24&25? that is less of a deprecation and essentially sunsetting/removing that functionality, that's why I was wary of doing that.

Reply:
Yes, please remove line 24 and 25.